libsignify/src/key.rs

```diff
@@ -162,14 +162,35 @@ impl PrivateKey {
                     return Err(Error::BadPassword);
                 }
 
-                // Confirmed the decryption worked, mutating the key structure.
-                self.complete_key = decrypted_key.0;
+                match PrivateKey::verify_keypair(&decrypted_key.0) {
+                    Ok(()) => {
+                        // Confirmed the decryption worked and the keys are matching.
+                        self.complete_key = decrypted_key.0;
+                    }
+                    Err(e) => return Err(e),
+                }
+
```

Comment:
This could also just be the following, couldn't it?

```suggestion
                PrivateKey::verify_keypair(&decrypted_key.0)?;
                // Confirmed the decryption worked and the keys are matching.
                self.complete_key = decrypted_key.0;
```

Reply:
Yeah that looks better, will switch to it 👍